mautic/mautic
# Use the device model when creating a new device so that the events are dispatched

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | 
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

We had LeadDevice events but they were not used because the new devices were persisted directly to the database via the repository. This fixes this so that the existing mautic.lead_device_* events are dispatched. 

#### Steps to test this PR:
1. Create a landing page and view it in an anonymous browser. Check that the new device was created for the contact. Hit the page again and ensure the same device ID is associated with the page hit in the database.
2. Send an email to a contact and open the email in an email client or private browser. Ensure the device was created in the database and associated with the stat in the email_stats_devices table.
